### PR TITLE
Drop support for Python 3.3

### PR DESCRIPTION
**Pull request by takluyver**
This makes 3.4 the minimum required version of Python. See discussion on #10161.

### CODE REVIEW

File: IPython/__init__.py [hunk 1]
```diff
@@ -27,12 +27,12 @@
 #-----------------------------------------------------------------------------
 
 # Don't forget to also update setup.py when this changes!
-if sys.version_info < (3,3):
+if sys.version_info < (3,4):
     raise ImportError(
 """
-IPython 6.0+ does not support Python 2.6, 2.7, 3.0, 3.1, or 3.2.
+IPython 6.3+ supports Python 3.4 and above.
```
Comment (hugovk) [inline IPython/__init__.py]: There's some similar text to update in setup.py

https://github.com/takluyver/ipython/blob/68559994a2c1efcbd408fa63adaf1867389e22df/setup.py#L47

Comment (takluyver) [reply IPython/__init__.py]: Thanks, updated.

**Review by willingc — APPROVED** [review]

**Comment by Carreau** [comment]
I'm Ok to drop 3.3 for 6.3; we just have to be careful, I don't think we've ever dropped a Python version a minor release though.

**Comment by willingc** [comment]
I think that we can provide guidance to pre-6.3 for anyone on Python 3.3.

**Comment by takluyver** [comment]
I'm also fine with leaving it until we split 6.x from master, and targetting it for 7.0. I don't think there's any hurry to do this - I don't know of anything we're eagerly waiting to do that's only possible from 3.4.

**Comment by Carreau** [comment]
I'm happy to wait a bit and see how much we are planning to keep the 6.x branch. We can fork master toward 7.0; though that might mean having a 5.x 6.x and master branch.

**Comment by takluyver** [comment]
I wouldn't start 7.0 for this; it's a simple set of changes, it's not worth having to backport everything for. Let's just let it sit until we have another breaking change that gives us a reason to aim for a 7.0 release.

**Review by hugovk — APPROVED** [review]

**Comment by jzf2101** [comment]
See #10911

**Comment by takluyver** [comment]
I think it's fine to stop testing with 3.3 in 6.x releases, and then make the changes that we know will break it for 7.0.

**Comment by Carreau** [comment]
> I think it's fine to stop testing with 3.3 in 6.x releases, and then make the changes that we know will break it for 7.0.

Ok, fair enough. This need rebase now. I think a nov/dec release would be welcome. Do we wait for Dec25 ?

**Comment by takluyver** [comment]
I've rebased the changes and updated the version numbers in various docs and messages to say that 7.0 (not 6.3) drops support for Python 3.3.

**Comment by Carreau** [comment]
@takluyver  do you want to merge this, so that we start master being a 7.0-dev ?

**Review by Carreau — APPROVED** [review]

**Comment by takluyver** [comment]
That's my plan. I've already updated the version number in master.

**Review by Carreau — APPROVED** [review]

**Pull request merged**